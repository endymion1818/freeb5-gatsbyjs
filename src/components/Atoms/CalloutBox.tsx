import React, { FC } from "react";
import styled from "styled-components";

import * as token from "../tokens";

export interface ICalloutBoxProps {
  /**
   * lighter design (on dark backgrounds)
   * @default false
   */
  light?: boolean;
  /**
   * text alignment
   * @default token.ETEXTALIGN.LEFT
   */
  textAlign?: token.ETEXTALIGN;
  /**
   * children of this component
   * @default <>test</>
   */
  children: string | JSX.Element[] | JSX.Element;
  /**
   * background colour
   * @default token.EBACKGROUND_COLOUR.SURFACE
   */
  backgroundColour?: token.EBACKGROUND_COLOUR;
  /**
   * text colour
   * @default token.ETEXT_COLOUR.ON_SURFACE
   */
  textColour?: token.ETEXT_COLOUR;
}

export interface ICalloutContainerProps extends ICalloutBoxProps {}

const CalloutContainer = styled.div<ICalloutContainerProps>`
  width: 80vw;
  max-width: 64rem;
  margin: 0 auto;
  padding: ${token.ESIZE.SINGLE};
  border-radius: 15px;

  ${({ textAlign }) => textAlign && `text-align: ${textAlign};`}

  background-color: ${({ backgroundColour }) =>
    backgroundColour ? backgroundColour : `rgba(0, 0, 0, 0.5);`};
  color: ${({ textColour }) =>
    textColour
      ? textColour
      : `${token.ETEXT_COLOUR.ON_SURFACE};
    text-shadow: 2px 2px 1px rgba(0,0,0,0.8);
    `};
`;

const CalloutBox: FC<ICalloutBoxProps> = ({
  children,
  light,
  textAlign,
  backgroundColour,
  textColour
}) => {
  if (!children) {
    return null;
  }
  return (
    <CalloutContainer
      light={light}
      textAlign={textAlign}
      backgroundColour={backgroundColour}
      textColour={textColour}
    >
      {children}
    </CalloutContainer>
  );
};

export default CalloutBox;
