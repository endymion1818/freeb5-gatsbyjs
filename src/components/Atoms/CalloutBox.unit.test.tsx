import { cleanup, render } from "@testing-library/react";
import React from "react";
import CalloutBox from "./CalloutBox";

describe("<CalloutBox />", () => {
  beforeEach(cleanup);
  it("Should render", () => {
    const { container } = render(<CalloutBox />);
    expect(container).toMatchSnapshot();
  });
});
