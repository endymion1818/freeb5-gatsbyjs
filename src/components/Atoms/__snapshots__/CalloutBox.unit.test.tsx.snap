// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalloutBox /> Should render 1`] = `
.c0 {
  width: 80vw;
  max-width: 64rem;
  margin: 0 auto;
  padding: 1rem;
  border-radius: 15px;
  background-color: rgba(0,0,0,0.5);
  color: white;
  text-shadow: 2px 2px 1px rgba(0,0,0,0.8);
}

<div>
  <div
    class="c0"
  >
    test
  </div>
</div>
`;
